plugins/inputs/consul/consul.go

```diff
@@ -45,6 +49,14 @@ var sampleConfig = `
   # tls_key = "/etc/telegraf/key.pem"
   ## Use TLS but skip chain & host verification
   # insecure_skip_verify = true
+
+  ## Gather all Consul checks' tags
+  # gather_all_tags = false
```

Comment:
I think we should remove this option and always gather all tags.  Plugin users can remove unwanted tags using [measurement filtering](https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#measurement-filtering).

Reply:
That is acctually not true. My app registers in consul and pass as tag uniq id. That is how it looks like now with telegraf 1.7.1
```
consul_health_checks_passing{="",
018ecd7a_4ca0_109d_cc67_45f4ba438280="",
03b93403_3960_49bb_81fc_8ebe0ff4c79a="",
03e9dd4a_b8de_48bf_ba98_050823dbe816="",
05cb54a7_1748_405b_9671_ccd630256364="",
06c7b6ac_9fd9_4494_ae3c_ac5d1daa9580="",
07b1163b_16ca_497b_afcc_2c9e6662f4ab="",
0b4c9bc8_78cf_407d_8a93_7d0d022cbab1="",
0b720e27_dc8a_e821_2638_08890c361783="",
0eaf2333_5929_4fc6_a304_c2452a05146a="",
10a7f37d_ff1b_4b90_b613_d3d500991355="",
135df3c5_822a_478c_a5e1_4fcdb9db23af="",
14e02d39_0afb_4e33_9a77_5d17a62b613b="",
```
So tags are random and i really don't know the way i exclude them from tags.
BTW there are also tag '' that makes my prometheus sad with 
`prom_1          | level=warn ts=2019-01-04T12:03:29.464871059Z caller=scrape.go:835 component="scrape manager" scrape_pool=stage target=http://10.143.114.19:9273/metrics msg="append failed" err="expected label name, got \"EQUAL\""
`
Of cause i can setup taginclude but that have to be split responsibility between devs and ops.

Reply:
Can you open a new issue for this?

Reply:
#5522